src/main.cpp

```diff
@@ -440,7 +440,166 @@ void affect_nearby_friends_with_alarm(struct Thing *traptng)
 
 long apply_wallhug_force_to_boulder(struct Thing *thing)
 {
-  return _DK_apply_wallhug_force_to_boulder(thing);
+  // return _DK_apply_wallhug_force_to_boulder(thing);
+  unsigned short angle;
+  long collide;
+  unsigned short new_angle;
+  struct Coord3d pos2;
+  struct Coord3d pos;
+  struct ShotStats *shotst = &shot_stats[thing->model];
+  unsigned short speed = (unsigned short)shotst->speed;
+  pos.x.val = thing->mappos.x.val + ((speed * lbSinTable[thing->move_angle_xy]) >> 16);
+  int n = speed * lbCosTable[thing->move_angle_xy];
+  pos.y.val = thing->mappos.y.val + (-(n >> 8) >> 8);
+  pos.z.val = thing->mappos.z.val;
+  game.action_rand_seed = ACTION_RANDOM(ULONG_MAX);
+  if ( !(game.action_rand_seed & 7) && (!thing->velocity.z.val ) )
+  {
+    if ( thing_touching_floor(thing) )
+    {
+      long top_cube = get_top_cube_at(thing->mappos.x.stl.num, thing->mappos.y.stl.num, NULL);
+      if ( ((top_cube & 0xFFFFFFFE) != 40) && (top_cube != 39) )
+      {
+        thing->veloc_push_add.z.val += 48;
+        thing->state_flags |= TF1_PushAdd;
+      }
+    }
+  }
+  if ( thing_in_wall_at(thing, &pos) )
+  {
+    long blocked_flags = get_thing_blocked_flags_at(thing, &pos);
+    if ( blocked_flags & SlbBloF_WalledX )
+    {
+      angle = thing->move_angle_xy;
+      if ( (angle) && (angle <= 1024) )
```

Comment:
512 1024 and 1536 are constants like ANGLE_90 ANGLE_180 and ANGLE_270
OR maybe DIRECTION_EAST and DIRECTION_SOUTH 
 (not important fix)